teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,290 @@
+
+To provide a consistent format across resolvers, we suggest the following `remote-resolver` template: `<resolver-name>:<type>/<namespace>/<name>@<version>#<subpath>?<additional-params>`.
+
+The `Remote Resolver` must define the format using the above `remote-resolver` template. The `remote-resolver` would be responsible for parsing the `resolver-reference` and extracting the necessary information. Tekton will simply validate that it is a valid `purl` spec. The requirements for the fields in the template are defined in the [purl](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) spec.
+
+If users provide an [invalid purl string](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component), a validation error will be thrown in the `TaskRun/PipelineRun's` `Status`.
+
+#### Git Resolver
+Based on `git` resolver's [params](https://tekton.dev/docs/pipelines/git-resolver/#parameters), we define the following purl spec:
+
+```
+git:<git-provider>/<org>/<repo>@<revision>#<pathInRepo>?token=<token>&tokenKey=<tokenKey>&serverURL=<serverURL>&scmType=<scmType>
+```
+
+Example:
+
+```
+git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: "git:github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token"
+```
+
+#### Bundle Resolver
+Based on `bundle` resolver's [params](https://tekton.dev/docs/pipelines/bundle-resolver/#parameters), we define the following purl spec:
+
+```
+bundle:<bundle-provider>/<bundle-namespace>/<bundle-name>@<bundle-version>?kind=<kind>&name=<name>&secret=<secret>
+``` 
+
+Example
+
+```
+bundle:docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: "bundle:docker.io/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default"
+```
+
+Occasionally, users may want to fetch from a private registry (e.g. at `10.96.190.208:5000`). This host contains a `:` and violates the requirement of a purl [`type`](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component) and would lead to an invalid `purl`. In cases where purl is violated, users cannot use the purl `resolver-reference`. They need to use the existing verbose way of providing the `resolver params`. Providing an invalid `purl` will lead to an error.  
```

Comment:
This is a pretty significant draw back - I think this needs to be in scope.

(same w/ http)

Reply:
Sounds good! Fixed.